### PR TITLE
Update everything

### PR DESCRIPTION
**Pull request by felixb**
* requires python 3.6 at least
* new formatted strings
* new api methods:
  * upload picture
  * get ad by id
  * get ad by id as xml
* fix api methods:
  * post xml in binary encoded utf8
* simplify code

### CODE REVIEW

**Comment by makefu** [comment]
make sure to add an example of how to create an ad with the new(old) library :)

**Comment by felixb** [comment]
Well, there is no change in the xml content at all.

**Comment by Smalakamp** [comment]
by creating an ad i always get the following error:

`<?xml version="1.0" encoding="UTF-8" standalone="yes"?><api-base-error xmlns="http://www.ebayclassifiedsgroup.com/schema/types/v1" http-status-code="400"><api-errors><api-error><message>Validation error: Null not allowed!</message></api-error></api-errors><api-debug-errors/><api-field-errors/></api-base-error>`

What am i doing wrong?

My XML looks like this:
<?xml version='1.0' encoding='UTF-8' standalone='yes' ?>
<ad:ad xmlns:types="http://www.ebayclassifiedsgroup.com/schema/types/v1" xmlns:cat="http://www.ebayclassifiedsgroup.com/schema/category/v1" xmlns:ad="http://www.ebayclassifiedsgroup.com/schema/ad/v1" xmlns:loc="http://www.ebayclassifiedsgroup.com/schema/location/v1" xmlns:attr="http://www.ebayclassifiedsgroup.com/schema/attribute/v1" xmlns:pic="http://www.ebayclassifiedsgroup.com/schema/picture/v1" xmlns:user="http://www.ebayclassifiedsgroup.com/schema/user/v1" xmlns:rate="http://www.ebayclassifiedsgroup.com/schema/rate/v1" xmlns:reply="http://www.ebayclassifiedsgroup.com/schema/reply/v1" xmlns:feed="http://www.ebayclassifiedsgroup.com/schema/feed/v1" locale="de_DE" id="0">
<ad:price>
	<types:currency-iso-code>
		<types:value localized-label="EUR">EUR</types:value>
	</types:currency-iso-code>
	
	<types:amount>99.00</types:amount>
	
	<types:price-type>
		<types:value>SPECIFIED_AMOUNT</types:value>
	</types:price-type>
</ad:price>

<ad:ad-type>
	<ad:value>OFFERED</ad:value>
</ad:ad-type>

<ad:ad-address>
	<types:state>Leipzig</types:state>
	<types:zip-code>04103</types:zip-code>
</ad:ad-address>

<ad:poster-type>
	<ad:value>PRIVATE</ad:value>
</ad:poster-type>

<ad:title>***</ad:title>	
<ad:description>***</ad:description>

<ad:email>***</ad:email>

<ad:contact-name>***</ad:contact-name>
<ad:user-id>***</ad:user-id>
<ad:seller-account-type>PRIVATE</ad:seller-account-type>
<ad:phone/>

<cat:category id="87"></cat:category>

<attr:attributes>
<attr:attribute name="sonstiges.versand"><attr:value>ja</attr:value></attr:attribute>
</attr:attributes>


<loc:locations><loc:location id="3878" />
</loc:locations>
</ad:ad>



Regards

**Comment by felixb** [comment]
@Smalakamp has this somthing to do with my PR? Or is this just a general question?

File: ebk_client.py [hunk 1]
```diff
@@ -145,61 +154,41 @@ def delete_ad(self, ad_id):
...
 
-    def get_location_by_coordinates(self, latitude, longitude, depth = None, include_parent_path = False):
-        url_suffix = "latitude={}&longitude={}".format(latitude,longitude)
-        locations = self.get_locations(url_suffix, depth, include_parent_path)
-        return locations
+    def get_location_by_name(self, location_name, depth=None, include_parent_path=False):
+        url_suffix = f'q={location_name}'
+        return self.get_locations(url_suffix, depth, include_parent_path)
 
-    def html_unescape(self, data):
```
Comment (tejado) [inline ebk_client.py:203]: As you removed this method, can you please fix example.py (line 47)?

Comment (felixb) [reply ebk_client.py:203]: Sorry. I missed that.
Due to minimum python version of 3.6 the whole try-import cascade is not necessary anymore.

**Comment by felixb** [comment]
I've not used any unescaping at all in my project. ;)

**Comment by makefu** [comment]
@felixb out of interest, would you mind sharing your project?
cheers!

**Pull request merged**